services/tables/src/main/java/com/linkedin/openhouse/tables/api/validator/impl/PoliciesSpecValidator.java

```diff
@@ -40,55 +41,106 @@ public class PoliciesSpecValidator {
   public boolean validate(
       Policies policies, TimePartitionSpec timePartitioning, TableUri tableUri, String schema) {
 
-    if (policies != null && policies.getRetention() != null) {
-      // Two invalid case for timePartitioned table
-      if (timePartitioning != null) {
-        if (policies.getRetention().getColumnPattern() != null) {
+    if (policies != null) {
```

Comment:
Is change in retention validation part of requirement? Would be good to keep this PR specific to replication specific change points. I think the validation logic for replication config can be cleanly added as a separate method.

Reply:
not required, I can change back this code refactor and add a new method